heft-plugins/heft-jest-plugin/src/HeftJestDataFile.ts

```diff
@@ -91,7 +69,7 @@ export class HeftJestDataFile {
    * Get the absolute path to the heft-jest-data.json file
    */
   public static getConfigFilePath(projectFolder: string): string {
-    return path.join(projectFolder, '.heft', 'build-cache', 'heft-jest-data.json');
+    return path.join(projectFolder, 'temp', 'heft-jest-data.json');
```

Comment:
What's the rationale behind this change? Is the idea that you don't want to cache the `.heft` folder, but you do want to cache the `temp` folder?

We need to come up with some formalism around temp folders and publish it somewhere, or just make it part of the Heft API. It may make sense to do the same with other folders (`lib`, `lib-*`, `src`, `dist`, `release`, `etc`, etc), although I don't think we need to do that right now.

Reply:
Also, this file needs to live in a directory that clearly associates it with its particular phase.

Reply:
The rationale is that the `.heft` folder will no longer exist in the new version of Heft. I spoke with Pete about this, and essentially came to the conclusion that we should:
- have a `.cache` folder where we keep stateful information that we can re-use in future runs/cache using Rush's caching
- have a `temp` folder where we store temporary information intended to be used within a single execution

Under both of these folders, plugins are given a `.cache/<unique folder>` and `temp/<unique folder>` in which they can store their information. These folder paths are gotten off of the `TaskSession` object that is passed to the plugin. I do like the idea of expanding to other folder types in the future, but for now this seems like a good starting point for the change.

Reply:
>plugins are given a `.cache/<unique folder>` and `temp/<unique folder>`

https://github.com/microsoft/rushstack/pull/3511